### ISSUE TITLE
RequestTimer and RequestLogger occasionally logged in the wrong threads

### ISSUE THREAD

**Issue by andrus** [bootique/bootique-jetty#84]
Was  analyzing log history for one of our production apps. The app is medium volume and is using `bootique-jetty-instrumented`. Logs for a "normal" request progression look like this:

```
bootique-http-73 5EKf9f0-80000002 i.b.j.i.r.RequestTimer: started
bootique-http-73 5EKf9f0-80000002 ... other request-specific logs here ... 
bootique-http-73 5EKf9f0-80000002 i.b.j.i.r.RequestTimer: finished in 12 ms
bootique-http-73 5EKf9f0-80000002 i.b.j.s.RequestLogger: 10.0.0.1 - - [08/Oct/2018:10:45:01 +0000] "GET / HTTP/1.1" 200 167 ...
```
But I am also seeing two cases that don't follow the pattern above. 

1. (Very common. Up to 15% of all requests) Missing `RequestLogger` line. Also all these requests seem to have nothing between start and finish log lines (no DB queries and such). So those may be some static resources (?)

```
bootique-http-41 5EKf9f0-80000001 i.b.j.i.r.RequestTimer: started
bootique-http-41 5EKf9f0-80000001 i.b.j.i.r.RequestTimer: finished in 4 ms
```

2. (Rare. Only saw it twice and on a single server) Transaction ID leaks beyond a "normal" transaction and covers two more requests on the same thread. In those extra requests `RequestLogger` line is logged without matching  `RequestTimer` lines. I guess unexpected absence of those leads to the TX ID not being reset. Not sure if this is relevant, but log rotation occurred between request 2 and 3 in all two occasions when I saw this issue happen.

```
bootique-http-16772 8-mXKoE-800023ca i.b.j.i.r.RequestTimer: started
bootique-http-16772 8-mXKoE-800023ca ... other request-specific logs here ... 
bootique-http-16772 8-mXKoE-800023ca i.b.j.i.r.RequestTimer: finished in 8 ms
bootique-http-16772 8-mXKoE-800023ca i.b.j.s.RequestLogger: 10.0.0.1 - - [06/Oct/2018:03:49:03 +0000] "GET /x HTTP/1.1" 200
bootique-http-16772 8-mXKoE-800023ca i.b.j.s.RequestLogger: 10.0.0.2 - - [06/Oct/2018:03:59:16 +0000] "GET /y HTTP/1.1" 200
bootique-http-16772 8-mXKoE-800023ca i.b.j.s.RequestLogger: 10.0.0.3 - - [06/Oct/2018:04:00:41 +0000] "GET /y HTTP/1.1" 200 
```

Number 1 is particularly troubling because it is such a big percentage of requests.

**Comment by andrus** [comment]
Not seeing this anymore on some of my BQ 1.1 apps with the latest Jetty. Perhaps that was a Jetty issue?

**Comment by andrus** [comment]
Closing. Haven't seen this in a while. Perhaps fixed by #104
